### PR TITLE
Correct hard coded platform used for building image so arm64 builds work

### LINKED ISSUE #839: AWS CCM DockerFile build for more than one platform
**What would you like to be added**:
The current Dockerfile for Cloud Provider AWS is primarily configured for the amd64 architecture. As the usage of different architectures (like ARM) is becoming more prevalent, especially in cloud-native environments, it would be beneficial for this project to support multi-platform Docker builds.

DockerFile reference https://github.com/kubernetes/cloud-provider-aws/blob/master/Dockerfile#L32

**Why is this needed**:
Supports users on different architectures, particularly important for ARM-based systems like AWS Graviton instance types. 

/kind feature

### PR DESCRIPTION
**Pull request by et304383**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Fixes #839 

**Which issue(s) this PR fixes**:
Fixes #839 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
NONE

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -29,7 +29,7 @@ ARG DISTROLESS_IMAGE=registry.k8s.io/build-image/go-runner:v2.3.1-go1.22.4-bookw
 ################################################################################
 # Build the manager as a statically compiled binary so it has no dependencies
 # libc, muscl, etc.
-FROM --platform=linux/amd64 ${GOLANG_IMAGE} as builder
+FROM --platform=${TARGETPLATFORM} ${GOLANG_IMAGE} as builder
```
Comment (hakman) [inline Dockerfile:32]: You may mean `BUILDPLATFORM`. For sure `TARGETPLATFORM` is incorrect.
AWS CCM already builds and publishes ARM64 images. Not exactly sure what you are trying to fix.
```suggestion
FROM --platform=${BUILDPLATFORM} ${GOLANG_IMAGE} as builder
```

**Review by hakman — CHANGES_REQUESTED** [review]

**Comment by et304383** [comment]
Closing as not required.

**Pull request closed without merging**